### PR TITLE
Prefs [TASK][26303] Handle Event for tracking screen

### PR DESCRIPTION
**Pull request by toandv98**
## Related Tickets

- [#Task 26303](https://edu-redmine.sun-asterisk.vn/issues/26303)

## WHAT

- ### Handle Event for tracking screen

## HOW

- Click DateRangePicker, click Check In Button, display data from local
- Handle check in, check out results, update local data

## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Reviewer1 (ToanDV) | Reviewer2 (name)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the source code have end-line errors?                   |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| Conventions   | No hard coding, use constants/configuration values           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |

## Notes (Optional)

*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/edu/sun/oereminder/data/repository/TimeSheetRepositoryImpl.kt [hunk 1]
```diff
@@ -19,19 +23,34 @@ class TimeSheetRepositoryImpl(
...
+    override fun getTimeRecord(date: Long, callback: SourceCallback<TimeRecord>) {
+        with(appExecutors) {
+            diskIO.execute {
+                val calendar = GregorianCalendar().from(date)
+                val timeRecords =
+                    local.getTimeRecords(calendar.firstMillisOfDay(), calendar.lastMillisOfDay())
+                        .firstOrNull()
+                mainThread.execute {
+                    if (timeRecords != null) callback.onSuccess(timeRecords)
+                    else callback.onError(Exception())
```
Comment (huytq-2422) [inline app/src/main/java/com/edu/sun/oereminder/data/repository/TimeSheetRepositoryImpl.kt]: Nếu không viết trên 1 dòng thì phải thêm ngoặc đầy đủ 
```suggestion
                    if (timeRecords != null){
                        callback.onSuccess(timeRecords)
                    } else {
                         callback.onError(Exception())
                     }
```

Comment (huytq-2422) [inline app/src/main/java/com/edu/sun/oereminder/data/repository/TimeSheetRepositoryImpl.kt]:
```suggestion
                    timeRecords?.let(callback::onSuccess)?: callback.onError(Exception())
```

File: app/src/main/java/com/edu/sun/oereminder/ui/timetracking/TimeTrackingPresenter.kt [hunk 2]
```diff
@@ -1,16 +1,78 @@
...
 
 class TimeTrackingPresenter(private val timeSheetRepository: TimeSheetRepository) :
     BasePresenterImpl<View>(), Presenter {
 
-    override fun load() {
+    private var isCheckIn = true
+    private var isShowFab = false
+    private var currentTimeRecord: TimeRecord? = null
+
+    override fun load() = timeSheetRepository.getTimeRecord(
```
Comment (huytq-2422) [inline app/src/main/java/com/edu/sun/oereminder/ui/timetracking/TimeTrackingPresenter.kt]:
```suggestion
    override fun đặt tên rõ ràng ra nhá () = timeSheetRepository.getTimeRecord(
```

File: app/src/main/java/com/edu/sun/oereminder/utils/DateTimeExtension.kt [hunk 3]
```diff
@@ -27,3 +28,32 @@ fun GregorianCalendar.firstDayOfMonth() = this.apply {
 fun GregorianCalendar.lastDayOfMonth() = this.apply {
     set(Calendar.DATE, getActualMaximum(Calendar.DATE))
 }
+
+fun GregorianCalendar.from(date: Long) = this.apply { timeInMillis = date }
+
+fun now() = GregorianCalendar().timeInMillis
```
Comment (thonc-2135) [inline app/src/main/java/com/edu/sun/oereminder/utils/DateTimeExtension.kt:34]: nên đổi tên khác, now là trạng từ

File: app/src/main/java/com/edu/sun/oereminder/utils/DateTimeExtension.kt [hunk 4]
```diff
@@ -27,3 +28,32 @@ fun GregorianCalendar.firstDayOfMonth() = this.apply {
...
+
+fun firstMillisOfMonth() = GregorianCalendar().firstDayOfMonth().timeInMillis
+
+fun lastMillisOfMonth() = GregorianCalendar().lastDayOfMonth().timeInMillis
+
+fun firstMillisOfDay() = GregorianCalendar().firstMillisOfDay()
+
+fun lastMillisOfDay() = GregorianCalendar().lastMillisOfDay()
```
Comment (thonc-2135) [inline app/src/main/java/com/edu/sun/oereminder/utils/DateTimeExtension.kt:42]: tương tự, k thì dùng anonymuoss funtion . Có thể lm theo hoặc không nhé

File: app/src/main/java/com/edu/sun/oereminder/ui/checkin/CheckInDialogFragment.kt [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            textCheckIn.text = getString(R.string.label_daily_check_out)
+            btnSendReport.text = getString(R.string.text_daily_report)
+            switchCheckIn.run {
+                setThumbResource(R.drawable.ic_close)
+                isChecked = true
+            }
+        }
+
+        btnBack.setOnClickListener { dismiss() }
+        btnSendReport.setOnClickListener {
```
Comment (thonc-2135) [inline app/src/main/java/com/edu/sun/oereminder/ui/checkin/CheckInDialogFragment.kt]: hình như t nhớ nên đỏi thành buttonBack, buttonSendReport

**Pull request merged**